test/unit/v1/connection.test.ts

```diff
@@ -0,0 +1,236 @@
+import { setupServer, SetupServerApi } from "msw/node";
+import { rest } from "msw";
+import { ConnectionOptions, Firebolt } from "../../../src";
+
+const apiEndpoint = "fake.api.com";
+
+const engineObject = {
+  id: {
+    engine_id: "123",
+    account_id: "some_account"
+  },
+  name: "some_engine",
+  endpoint: "https://some_engine.com"
+};
+
+const selectOneResponse = {
+  meta: [
+    {
+      name: "one",
+      type: "Int32"
+    }
+  ],
+  data: [
+    {
+      one: 1
+    }
+  ],
+  rows: 1
+};
+
+const defaultAccountResponse = {
+  account: {
+    id: "some_other_account",
+    name: "some_other_account_name",
+    title: "some_other_account_name"
+  }
+};
+
+function resetServerHandlers(server: SetupServerApi) {
+  server.use(
+    // Authentication
+    rest.post(`https://${apiEndpoint}/auth/v1/login`, (req, res, ctx) => {
+      return res(
+        ctx.json({
+          access_token: "fake_access_token",
+          refresh_token: "fake_refresh_token"
+        })
+      );
+    }),
+    // Get account id
+    rest.get(
+      `https://${apiEndpoint}/iam/v2/accounts:getIdByName`,
```

Comment:
It doesn't need to since otherwise it accepts all parameters. However, we sure can check that it was passed correctly. I've added an `expect` statement.